lib/unsplash/connection.rb

```diff
@@ -127,7 +127,9 @@ def request(verb, path, params = {})
 
       status_code = response.respond_to?(:status) ? response.status : response.code
 
-      if !(200..299).include?(status_code)
+      if status_code >= 500
+        raise Unsplash::Error.new response.body
+      elsif !(200..299).include?(status_code)
         body = JSON.parse(response.body)
         msg = body["error"] || body["errors"].join(" ")
         raise Unsplash::Error.new msg
```

Comment:
I wonder if this could be more generic by rescuing the `JSON::ParserError` and raising the `response.body` then? Then it would cover other status codes between 300 and 499 =)

For example:
```ruby
begin
  if !(200..299).include?(status_code)
    body = JSON.parse(response.body)
    msg = body["error"] || body["errors"].join(" ")
    raise Unsplash::Error.new msg
  end
rescue JSON::ParserError
  raise Unsplash::Error.new response.body
end
```

Reply:
I think that your proposed solution is much cleaner, I guess I got tunnel-visioned to verifying the status code is between two integers like the `200..299` conditional. :)
Fixed!

Reply:
Thanks so much! Fresh eyes :eyes: